core/system/src/unvic.c

```diff
@@ -91,16 +93,11 @@ static int unvic_acl_check(int irqn)
     /* get vector entry */
     uv = &g_unvic_vector[irqn];
 
-    /* an IRQn slot is considered as registered if the ISR entry in the unprivileged
-     * table is not 0 */
-    is_irqn_registered = uv->hdlr ? 1 : 0;
+    /* an IRQn slot is considered as registered if the vector id is not unused. */
+    is_irqn_registered = (uv->id != UNVIC_VECTOR_UNUSED);
```

Comment:
We wouldn't need this check if, instead of `UNVIC_VECTOR_UNUSED`, we use an invalid box ID define (which we may need to create). Simply checking `uv->id != g_active_box` would be good enough.